src/library_async.js

```diff
@@ -385,12 +385,14 @@ mergeInto(LibraryManager.library, {
     Asyncify.handleSleep(function(wakeUp) {
       // We must first unwind, so things are spilled to the stack. We
       // can resume right after unwinding, no need for a timeout.
```

Comment:
the comment here is now outdated. I think now we need to explain here why the Promise is needed - to avoid unbounded call stack growth, etc.